[PTD][BE][c10d] Add some code documents for TCPStore code and cosmetic changes to libUVStore code

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130496

While designing something else when TCPStore is needed. I spent some time digging into the codebase of TCPStore and found that the code is a little bit challenging to understand without proper documents. Although people from OSS community must be smarter than me, I still want to document my findings in the code so that devs and users can use them as a reference down the road.

Also for libuv, we need to make private variables with a "_", so it's a pure renaming of private variables such as `tcpServer`.



cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

fduwjj committed:
[PTD][c10d] Add some document for TCPStore code and cosmetic changes to libuvStore code

[ghstack-poisoned]

fduwjj committed:
Update on "[PTD][c10d] Add some document for TCPStore code and cosmetic changes to libuvStore code"


While designing something else when TCPStore is needed. I spent some time digging into the codebase of TCPStore and found that the code is a little bit challenging to understand without proper documents. Although people from OSS community must be smarter than me, I still want to document my findings in the code so that devs and users can use them as a reference down the road.

Also for libuv, we need to make private variable with a "_", so it's a pure renaming of a private variable for `tcpServer`.



cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj committed:
Update on "[PTD][BE][c10d] Add some code documents for TCPStore code and cosmetic changes to libUVStore code"


While designing something else when TCPStore is needed. I spent some time digging into the codebase of TCPStore and found that the code is a little bit challenging to understand without proper documents. Although people from OSS community must be smarter than me, I still want to document my findings in the code so that devs and users can use them as a reference down the road.

Also for libuv, we need to make private variable with a "_", so it's a pure renaming of a private variable for `tcpServer`.



cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

wconstab commented on torch/csrc/distributed/c10d/TCPStore.hpp:
```diff
@@ -9,6 +9,33 @@
 namespace c10d {
 namespace detail {
 
+// TCPStore essentially is a key-value pair storage we used for rendezvous
```
TCPStore is a key-value store used by PyTorch mainly for distributed rendezvous, but for other purposes as well.

wconstab commented on torch/csrc/distributed/c10d/TCPStore.hpp:
```diff
@@ -9,6 +9,33 @@
+//
+// Code structure:
+// ├── TCPStore client side API and server setup code:
+// │   TCPStore.hpp/TCPStore..cpp
```
duplicate ..

fduwjj replied:
opps, Let me clean it up..

fduwjj committed:
Update on "[PTD][BE][c10d] Add some code documents for TCPStore code and cosmetic changes to libUVStore code"


While designing something else when TCPStore is needed. I spent some time digging into the codebase of TCPStore and found that the code is a little bit challenging to understand without proper documents. Although people from OSS community must be smarter than me, I still want to document my findings in the code so that devs and users can use them as a reference down the road.

Also for libuv, we need to make private variables with a "_", so it's a pure renaming of private variables such as `tcpServer`.



cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge

wz337 commented on torch/csrc/distributed/c10d/TCPStore.hpp:
```diff
@@ -9,6 +9,33 @@
 namespace c10d {
 namespace detail {
 
+// TCPStore is a key-value store used by PyTorch mainly for distributed
```
Appreciate great documentation. I am learning just by reading this nice doc.

fduwjj committed:
Update on "[PTD][BE][c10d] Add some code documents for TCPStore code and cosmetic changes to libUVStore code"


While designing something else when TCPStore is needed. I spent some time digging into the codebase of TCPStore and found that the code is a little bit challenging to understand without proper documents. Although people from OSS community must be smarter than me, I still want to document my findings in the code so that devs and users can use them as a reference down the road.

Also for libuv, we need to make private variables with a "_", so it's a pure renaming of private variables such as `tcpServer`.



cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj commented:
Somehow this PR did not get merged... Let's merge it now.

fduwjj commented:
@pytorchbot merge -i
